src/Adapter/MSTest.TestAdapter/Discovery/TypeValidator.cs

```diff
@@ -106,7 +106,7 @@ internal static bool HasCorrectTestContextSignature(Type type)
     {
         DebugEx.Assert(type != null, "HasCorrectTestContextSignature type is null");
 
-        IEnumerable<PropertyInfo> propertyInfoEnumerable = type.GetTypeInfo().DeclaredProperties;
+        IEnumerable<PropertyInfo> propertyInfoEnumerable = type.GetProperties(BindingFlags.DeclaredOnly);
```

Comment:
This is not equivalent. According to documentation `DeclaredProperties` returns properties from hierarchy while `BindingFlags.DeclaredOnly` returns only properties of current type.

I would need to make some tests to see if `static` are considerd by `DeclaredProperties` or not to see if that's equivalent to `GetProperties()`.